pkgs/development/tools/rye/default.nix

```diff
@@ -9,24 +11,40 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "rye";
-  version = "unstable-2023-04-23";
+  version = "0.5.0";
 
-  src = fetchFromGitHub {
-    owner = "mitsuhiko";
-    repo = "rye";
-    rev = "b3fe43a4e462d10784258cad03c19c9738367346";
-    hash = "sha256-q9/VUWyrP/NsuLYY1+/5teYvDJGq646WbMXptnUUUyw=";
+  src = fetchgit {
+    url = "https://github.com/mitsuhiko/${pname}.git";
+    rev = version;
+    # One of the tests runs git command so we need .git directory there
```

Comment:
I'm not sure but possibly it is worth to disable a failed test. Here is it, the one that requires git repository:

```
failures:

---- utils::test_is_inside_git_work_tree::test_is_inside_git_work_tree_true stdout ----
thread 'utils::test_is_inside_git_work_tree::test_is_inside_git_work_tree_true' panicked at 'assertion failed: is_inside_git_work_tree(&PathBuf::from(\".\"))', rye/src/utils.rs:493:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Yes, you can either disable this specific test
```nix
cargoTestFlags = [
  "-- --skip=utils::test_is_inside_git_work_tree::test_is_inside_git_work_tree_true"
];
```

or keep the `.git` folder. For this, you can still use `fetchFromGithub` and use `leaveDotGit`.